File: dev-superior-algoritmo-e-logica-de-programacao/java/src/exercicios/Comerciante.java
package exercicios;

import java.util.Locale;
import java.util.Scanner;

public class Comerciante {

	public static void main(String[] args) {

		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);

		int N, abaixo, entre, acima;
		double totalCompra, totalVenda, totalLucro, lucro, percentualLucro;

	    System.out.print("Serao digitados dados de quantos produtos? ");
		N = sc.nextInt();

	    String[] nomes = new String[N];
	    double[] precosCompra = new double[N];
	    double[] precosVenda = new double[N];

		for (int i = 0; i < N; i++) {
	    	System.out.printf("Produto %d:\n", i + 1);
	    	System.out.print("Nome: ");
	    	sc.nextLine();
	        nomes[i] = sc.nextLine();
	        System.out.print("Preco de compra: ");
			precosCompra[i] = sc.nextDouble();
	        System.out.print("Preco de venda: ");
			precosVenda[i] = sc.nextDouble();
		}

		abaixo = 0;
		entre = 0;
		acima = 0;
		for (int i = 0; i < N; i++) {
			lucro = precosVenda[i] - precosCompra[i];
			percentualLucro = lucro * 100.0 / precosCompra[i];

			if (percentualLucro < 10.0) {
				abaixo++;
			} else if (percentualLucro < 20.0) {
				entre++;
			} else {
				acima++;
			}
		}

		totalCompra = 0;
		totalVenda = 0;
		for (int i = 0; i < N; i++) {
			totalCompra = totalCompra + precosCompra[i];
			totalVenda = totalVenda + precosVenda[i];
		}

		totalLucro = totalVenda - totalCompra;

	    System.out.println("\nRELATORIO:");
	    System.out.printf("Lucro abaixo de 10%%: %d\n", abaixo);
	    System.out.printf("Lucro entre 10%% e 20%%: %d\n", entre);
	    System.out.printf("Lucro acima de 20%%: %d\n", acima);
	    System.out.printf("Valor total de compra: %.2f\n", totalCompra);
	    System.out.printf("Valor total de venda: %.2f\n", totalVenda);
	    System.out.printf("Lucro total: %.2f\n", totalLucro);

		sc.close();
	}
}

File: nelio-alves-algoritmo-e-logica-de-programacao/python/exercicios/comerciante.py
N: int; abaixo: int; entre: int; acima: int
totalCompra: float; totalVenda: float; totalLucro: float; lucro: float; percentualLucro: float

N = int(input("Serao digitados dados de quantos produtos? "))

nomes: [str] = [0 for x in range(N)]
precosCompra: [float] = [0 for x in range(N)]
precosVenda: [float] = [0 for x in range(N)]

for i in range(N):
	print(f"Produto {i + 1}:")
	nomes[i] = str(input("Nome: "))
	precosCompra[i] = float(input("Preco de compra: "))
	precosVenda[i] = float(input("Preco de venda: "))

abaixo = 0
entre = 0
acima = 0
for i in range(N):
	lucro = precosVenda[i] - precosCompra[i]
	percentualLucro = lucro * 100.0 / precosCompra[i]

	if percentualLucro < 10.0:
		abaixo = abaixo + 1
	elif percentualLucro < 20.0:
		entre = entre + 1
	else:
		acima = acima + 1

totalCompra = 0
totalVenda = 0
for i in range(N):
	totalCompra = totalCompra + precosCompra[i]
	totalVenda = totalVenda + precosVenda[i]

totalLucro = totalVenda - totalCompra
print("\nRELATORIO:")
print(f"Lucro abaixo de 10%: {abaixo}")
print(f"Lucro entre 10% e 20%: {entre}")
print(f"Lucro acima de 20%: {acima}")
print(f"Valor total de compra: {totalCompra:.2f}")
print(f"Valor total de venda: {totalVenda:.2f}")
print(f"Lucro total: {totalLucro:.2f}")